pkg/operator/resources/resources.go

```diff
@@ -273,3 +341,22 @@ func namesAndIDsFromStatuses(statuses []status.Status) ([]string, []string) {
 
 	return apiNames, apiIDs
 }
+
+func checkIfUsedByAPISplitter(apiName string) error {
+	apiRes, err := GetAPIs()
```

Comment:
GetAPIs can be a more expensive call that is required. It looks like we can figure out is an APISplitter is using this function based on routes in virtualservices that have the label apiKind: APISplitter. You can use k8s.ListVirtualServicesByLabels to find the relevant virtual services and then reconstruct which apis are being used.

Reply:
good idea